palantir-java-format/src/test/resources/com/palantir/javaformat/java/testdata/RSL.output

```diff
@@ -1,6 +1,106 @@
 class RSLs {
-    String s = """
-    lorem
-      ipsum
-    """;
+    String a = """
+        lorem
+          ipsum
+        """;
+    String b = """
+        lorem
+        ipsum
+        """;
+    String c = """
+        lorem
+        ipsum
+        """;
+    String d = """
+        ipsum
+        """;
+    String e = """
+        """;
+    String f = """
+        ipsum\
+        """;
+    String g = """
+        lorem\
+        ipsum
+        """;
+    String h = """
+        lorem\
+        ipsum\
+        """;
+    String i = """
+        lorem
+
+        ipsum
+        """;
+    String j = """
+        lorem
+        one long incredibly unbroken sentence moving from topic to topic so that no one had a chance to interrupt
+        ipsum
+        """;
+    String k = """
+        lorem
+        ipsum
+        """;
+    String l = """
+          foo
+        bar
+          baz\
+        """;
+
+    {
+        f("""
+        lorem
+        ipsum
+        """, 42);
+
+        """
+        hello %s
+        """.formatted("world");
+        f(/* foo= */ """
+            foo
+            """, /* bar= */ """
+            bar
+            """);
+        """
+        hello
+        """.codePoints().forEach(System.err::println);
+        String s = """
+            foo
+            """ + """
+            bar
+            """;
+        // TODO(crogoz): fix formatSource rendering of this block
+        String broken = """
+            foo
+            """ + """
+        bar
+        """;
+        String working = """
+            foo
+            """ + """
+            bar
+            """;
+        String u = stringVariableOne + """
```

Comment:
unfortunately string concatenation is not great with text blocks and I am not sure if I can change it. 
after running the AST formatter I am getting (The AST cannot change the indentation of the text block node): 
```
    String u = stringVariableOne + """
            ...
            """ + stringVariableTwo + """
                    ...
                    """;
```
and then after I try to re-indent the text blocks and given that I am trying to indent based on the previous line, I am getting the block below. 

For these kind of text block operations, people should try to keep the textBlocks indented at the same line with the rest of the operands and then the block will be properly indented